lib/lists.nix

```diff
@@ -1688,17 +1688,22 @@ rec {
     ## `lib.lists.crossLists` usage example
 
     ```nix
-    crossLists (x:y: "${toString x}${toString y}") [[1 2] [3 4]]
+    crossLists (x: y: "${toString x}${toString y}") [[1 2] [3 4]]
     => [ "13" "14" "23" "24" ]
     ```
 
+    To migrate to `lib.mapCartesianProduct` we need to pass an attrset as input instead of a list:
+
+    ```nix
+    mapCartesianProduct (x: "${toString x.a}${toString x.b}") { a = [1 2]; b = [3 4]; }
+    => [ "13" "14" "23" "24" ]
+    ```
     :::
   */
   crossLists = warn
-    "lib.crossLists is deprecated, use lib.cartesianProductOfSets instead."
+    "lib.crossLists is deprecated, use lib.mapCartesianProduct instead."
```

Comment:
I guess if we want to actually make sure that people understand how to use a new function, this should come with an example. Though in this case, `crossLists` is already deprecated for a while.

Reply:
The updated documentation shows how to migrate from `crossLists` to `mapCartesianProduct`, is it usually in both places (the warn message and the docs)? Happy to add it, it's been a while since I've contributed to nixpkgs :D

Reply:
If that's the case, do we use something like this for multi-line warning messages (using double `'` single quotes)?

```nix
 crossLists = warn ''
  lib.crossLists is deprecated, use lib.mapCartesianProduct instead.

  To migrate to `lib.mapCartesianProduct` we need to pass an attrset as input instead of a list:

  ```nix
  mapCartesianProduct (x: "${toString x.a}${toString x.b}") { a = [1 2]; b = [3 4]; }
  => [ "13" "14" "23" "24" ]
  \```
''
```

Reply:
@infinisil sounds good, but where? Both in the documentation and the warn message? One or the other?

Reply:
Added an example in both places. Here's how it looks in the REPL:

![crossLists-deprecation](https://github.com/NixOS/nixpkgs/assets/443978/5bd6305a-eda8-44b9-afca-020b311bb465)

Let me know if you're happy with it, I'll proceed with getting the commits right as suggested.

Reply:
Sounds good to me!